tests/vmi_cloudinit_test.go

```diff
@@ -258,54 +260,44 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 			})
 		})
 
-		Context("with cloudInitNoCloud userData source", func() {
-			It("[test_id:1617]should process provided cloud-init data", func() {
-				userData := fmt.Sprintf("#!/bin/sh\n\necho '%s'\n", expectedUserData)
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), userData)
-
-				vmi = LaunchVMI(vmi)
-				CheckCloudInitIsoSize(vmi, cloudinit.DataSourceNoCloud)
-				By("executing a user-data script")
-				VerifyUserDataVMI(vmi, []expect.Batcher{
-					&expect.BSnd{S: "\n"},
-					&expect.BExp{R: expectedUserData},
-				}, time.Second*120)
+		Context("should process provided cloud-init data", func() {
+			userData := fmt.Sprintf("#!/bin/sh\n\ntouch /%s\n", expectedUserData)
 
-				By("applying the hostname from meta-data")
-				Expect(console.LoginToCirros(vmi)).To(Succeed())
+			runTest := func(vmi *v1.VirtualMachineInstance, dsType cloudinit.DataSourceType) {
+				vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
 
-				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
-					&expect.BSnd{S: "hostname\n"},
-					&expect.BExp{R: dns.SanitizeHostname(vmi)},
-				}, 10)).To(Succeed())
-			})
-		})
-
-		Context("with cloudInitConfigDrive userData source", func() {
-			It("[test_id:3180]should process provided cloud-init data", func() {
-				userData := fmt.Sprintf("#!/bin/sh\n\necho '%s'\n", expectedUserData)
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndConfigDriveUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), userData)
+				By("waiting until login appears")
+				libwait.WaitUntilVMIReady(vmi, console.LoginToCirros)
 
-				vmi = LaunchVMI(vmi)
-				CheckCloudInitIsoSize(vmi, cloudinit.DataSourceConfigDrive)
-				By("executing a user-data script")
-				VerifyUserDataVMI(vmi, []expect.Batcher{
-					&expect.BSnd{S: "\n"},
-					&expect.BExp{R: expectedUserData},
-				}, time.Second*120)
+				By("validating cloud-init disk is 4k aligned")
+				CheckCloudInitIsoSize(vmi, dsType)
 
-				By("applying the hostname from meta-data")
-				Expect(console.LoginToCirros(vmi)).To(Succeed())
+				By("Checking whether the user-data script had created the file")
+				Expect(console.RunCommand(vmi, fmt.Sprintf("cat /%s\n", expectedUserData), time.Second*120)).To(Succeed())
 
+				By("validating the hostname matches meta-data")
 				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
 					&expect.BSnd{S: "hostname\n"},
 					&expect.BExp{R: dns.SanitizeHostname(vmi)},
 				}, 10)).To(Succeed())
+			}
+
+			It("[test_id:1617] with cloudInitNoCloud userData source", func() {
+				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(
+					cd.ContainerDiskFor(cd.ContainerDiskCirros),
+					userData)
+				runTest(vmi, cloudinit.DataSourceNoCloud)
```

Comment:
Nit: couldn't you use a `DescribeTable` for this?

Reply:
`cd.ContainerDiskFor` can panic therefore I cannot call it from `Entry`, and I wanted to avoid using libvmi in that case because there are some issues with the Cirros builder and adding more cloud-init disks.